donut/__init__.py

```diff
@@ -151,6 +151,23 @@ def access_forbidden(error):
     return flask.render_template("403.html"), http.client.FORBIDDEN
 
 
+@app.errorhandler(http.client.REQUEST_ENTITY_TOO_LARGE)
+def upload_too_large(error):
+    """ Handles a 413 Payload Too Large error. """
+    UNITS = ('B', 'KB', 'MB', 'GB', 'TB')
+    next_max_size = constants.MAX_CONTENT_LENGTH
+    for unit in UNITS:
+        max_size = next_max_size
+        next_max_size /= 1024
+        if next_max_size < 1:
+            break
+    int_max_size = int(max_size)
+    if int_max_size == max_size:
+        max_size = int_max_size
+    return flask.render_template("413.html", max_size=f'{max_size} {unit}'), \
```

Comment:
Why not just `return flask.render_template("413.html", max_size='10 MB')` ? What's the point of the code above?

Reply:
I was trying to avoid having to specify the limit in two different places, but I agree this unit computation is a bit overkill. Made the human-readable version of the size a constant next to `MAX_CONTENT_LENGTH` in the constants file.